#!/usr/bin/env python
PACKAGE = "hk_camera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Exposure
gen.add("exposure_value", double_t, 0, "Exposure time when not in auto mode", 100., 20., 30000.)
gen.add("auto_exposure_min", double_t, 0, "Min exposure time(us)", 100., 20., 30000.)
gen.add("auto_exposure_max", double_t, 0, "Max exposure time(us)", 5000., 20., 30000.)
gen.add("exposure_auto", bool_t, 0, "Auto exposure", True)

# Gain
gen.add("gain_value", double_t, 0, "Gain value(dB) when not in auto mode", 2., 0., 16.)
gen.add("auto_gain_min", double_t, 0, "Min gain dB", 0., 0., 16.)
gen.add("auto_gain_max", double_t, 0, "Max gain dB", 16., 0., 16.)
gen.add("gain_auto", bool_t, 0, "Auto Gain", True)

# White balance
gen.add("balance_white_auto", bool_t, 0, "Auto white", True)

# Gamma Notice cs050 and cs020 is not suppose this config!!!
gamma_selector = gen.enum([gen.const("Gamma_auto", int_t, 0, "Gamma auto"),
                           gen.const("Gamma_user", int_t, 1, "Gamma user"),
                           gen.const("Gamma_off", int_t, 2, "Gamma off")],
                           "gamma_selector")
gen.add("gamma_selector", int_t, 0, "Gamma selector", 0, 0, 2, edit_method = gamma_selector)
gen.add("gamma_value", double_t, 0, "Gamma value", 0.7, 0., 4.)

exit(gen.generate(PACKAGE, "hk_camera", "Camera"))
